commands/addons/wait.js

```diff
@@ -20,15 +19,60 @@ function * run (ctx, api) {
   }
 }
 
+function anyProvisioning (addons) {
+  return addons.some((addon) => {
+    return (addon.state === 'provisioning')
+  })
+}
+
+function * waitForMany (ctx, api, app, interval) {
+  yield cli.action(`Waiting for add-ons to be created on ${cli.color.app(app)}`, co(function * () {
+    let addons, formattedAddons
+    let stillProvisioning = true
+
+    while (stillProvisioning) {
+      addons = yield api.get(`/apps/${app}/addons`, {headers: {'Accept-Expansion': 'addon_service,plan'}})
+
+      if (isTTY) {
+        formattedAddons = addons.map((addon) => {
+          return `${cli.color.addon(addon.name)} (${addon.plan.name}) ${formatState(addon.state)}`
+        }).join('\n')
+
+        cli.action.status('\n' + formattedAddons)
+      }
+
+      yield wait(interval * 1000)
+
+      stillProvisioning = anyProvisioning(addons)
+    }
+  }))
+}
+
+function * run (ctx, api) {
+  const resolve = require('../../lib/resolve')
+
+  let interval = parseInt(ctx.flags['wait-interval'])
+  if (!interval || interval < 0) { interval = 5 }
+
+  if (ctx.args.addon) {
+    let addon = yield resolve.addon(api, ctx.app, ctx.args.addon, {'Accept-Expansion': 'addon_service,plan'})
+    yield waitForOne(ctx, api, addon, interval)
+  } else {
+    yield waitForMany(ctx, api, ctx.app, interval)
+  }
+}
+
 let topic = 'addons'
 
 module.exports = {
   topic: topic,
   command: 'wait',
   wantsApp: true,
   needsAuth: true,
-  args: [{name: 'addon'}],
-  flags: [{name: 'wait-interval', description: 'how frequently to poll in seconds', hasValue: true}],
+  args: [{name: 'addon', optional: true}],
+  flags: [
+    {name: 'wait-interval', char: 'i', description: 'how frequently to poll in seconds', hasValue: true}
```

Comment:
I feel like this could be just `--interval` instead of `--wait-interval` because the command is already `wait`. It's already clear what the interval if for.